src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -41,69 +46,108 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
+
+                        // Calls for a multiple null check only when this is the last itieration and there's more than one possible valid parameter parameter
+                        if (listOfValidParametersNodes.Count > 1 && counter == listOfParameterNodes.Count())
+                        {
+                            context.RegisterRefactorings(await GetRefactoringsAsync(
+                                document, functionDeclaration, methodSymbol, blockStatementOpt, listOfValidParametersNodes, position, cancellationToken).ConfigureAwait(false));
+                        }
+                    }
+                }
+                if (!checkIfToReturn)
                 {
                     return;
                 }
             }
+        }
 
-            var functionDeclaration = parameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
-            if (functionDeclaration == null)
-            {
-                return;
-            }
-
-            var parameterDefault = syntaxFacts.GetDefaultOfParameter(parameterNode);
-
-            // Don't offer inside the "=initializer" of a parameter
-            if (parameterDefault?.Span.Contains(position) == true)
-            {
-                return;
-            }
-
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-
-            // we can't just call GetDeclaredSymbol on functionDeclaration because it could an anonymous function,
-            // so first we have to get the parameter symbol and then its containing method symbol
-            var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
-            if (parameter == null || parameter.Name == "")
-            {
-                return;
-            }
-
-            var method = (IMethodSymbol)parameter.ContainingSymbol;
-            if (method.IsAbstract ||
-                method.IsExtern ||
-                method.PartialImplementationPart != null ||
-                method.ContainingType.TypeKind == TypeKind.Interface)
-            {
-                return;
-            }
-
-            if (CanOfferRefactoring(functionDeclaration, semanticModel, syntaxFacts, cancellationToken, out var blockStatementOpt))
-            {
-                // Ok.  Looks like a reasonable parameter to analyze.  Defer to subclass to 
-                // actually determine if there are any viable refactorings here.
-                context.RegisterRefactorings(await GetRefactoringsAsync(
-                    document, parameter, functionDeclaration, method, blockStatementOpt, cancellationToken).ConfigureAwait(false));
-            }
+        protected virtual Task<ImmutableArray<CodeAction>> GetRefactoringsAsync(Document document, SyntaxNode functionDeclaration, IMethodSymbol method, IBlockOperation blockStatementOpt, IEnumerable<SyntaxNode> listOfParameterNodes, int position, CancellationToken cancellationToken)
```

Comment:
Consider renaming this to something like `GetRefactoringsForAllParametersAsync`.